### PR TITLE
feat(lyrics-plus): chinese characters convertion

### LINKED ISSUE #2174: Possible to add translation of Chinese characters
### Chinese characters conversation in the Lyrics Plus APP

Thank you Spicetify & Lyrics Plus developers for making such great extension!

For the Lyrics Plus App,

In Chinese, we have traditional/simplified characters. When a Hong Kong Cantonese song is being played, sometimes the Lyric Plus extension will fetch a Simplified Version of Chinese lyrics. It may be possible to add a translation button to automatically convert Simplified Characters (used in Mainland China) to Traditional Characters (Used in Hong Kong/ Taiwan).

There is already a "[opencc-js](https://github.com/nk2028/opencc-js)" on GitHub which easily convert Chinese characters w/ a JS.
For Traditional to Simplified Characters: https://cdn.jsdelivr.net/npm/opencc-js@1.0.5/dist/umd/t2cn.js
For Simplified to Traditional: https://cdn.jsdelivr.net/npm/opencc-js@1.0.5/dist/umd/cn2t.js
Full conversation JS: https://cdn.jsdelivr.net/npm/opencc-js@1.0.5/dist/umd/full.js

I believe this feature will greatly increase the functionality of the Lyrics Plus app.

### ➕ Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Deskehs**
Add support for Chinese Language conversion without interrupting Japanese Language conversion. Closes #2174 

https://user-images.githubusercontent.com/124361319/226099583-afd336ab-6338-4d8d-bc64-39816acb6be9.mp4

https://user-images.githubusercontent.com/124361319/226099924-35b73582-d8f4-42c8-ab8d-1e376a08bfd5.mp4

### CODE REVIEW

File: CustomApps/lyrics-plus/index.js [hunk 1]
```diff
@@ -292,20 +310,26 @@ class LyricsContainer extends react.Component {
 			["hiragana", "furigana", "furigana"],
 			["hiragana", "normal", "hiragana"],
 			["katakana", "normal", "katakana"]
-		].map(params =>
+		].map(params => {
+			if (language != "ja") return;
```
Comment (rxri) [inline CustomApps/lyrics-plus/index.js:314]:
```suggestion
			if (language !== "ja") return;
```

File: CustomApps/lyrics-plus/index.js [hunk 2]
```diff
@@ -292,20 +310,26 @@ class LyricsContainer extends react.Component {
...
+			});
+		});
+		[
+			["cn", "hk"],
+			["cn", "tw"],
+			["t", "cn"],
+			["t", "hk"],
+			["t", "tw"]
+		].map(params => {
+			if (!language.includes("zh") || (language == "zh-hans" && params[0] == "t") || (language == "zh-hant" && params[0] == "cn")) return;
```
Comment (rxri) [inline CustomApps/lyrics-plus/index.js:327]:
```suggestion
			if (!language.includes("zh") || (language === "zh-hans" && params[0] === "t") || (language === "zh-hant" && params[0] === "cn")) return;
```

File: CustomApps/lyrics-plus/Translator.js [hunk 3]
```diff
@@ -43,15 +46,17 @@ class Translator {
 		};
 	}
 
-	async createKuroshiro() {
-		if (typeof Kuroshiro === "undefined" || typeof KuromojiAnalyzer === "undefined") {
+	async createTranslator() {
+		if (typeof Kuroshiro === "undefined" || typeof KuromojiAnalyzer === "undefined" || typeof OpenCC === "undefined") {
 			//Waiting for JSDeliver to load Kuroshiro and Kuromoji
```
Comment (rxri) [inline CustomApps/lyrics-plus/Translator.js:51]:
```suggestion
			// Waiting for JSDeliver to load Kuroshiro, Kuromoji and OpenCC
```

**Review by rxri — CHANGES_REQUESTED** [review]

File: CustomApps/lyrics-plus/Settings.js [hunk 4]
```diff
@@ -484,6 +484,15 @@ function openConfig() {
 					min: thresholdSizeLimit.min,
 					max: thresholdSizeLimit.max,
 					step: thresholdSizeLimit.step
+				},
+				{
+					desc: "Text convertion: Traditional-Simplified Detection threshold (Advanced)",
+					info: "Same as above but with Traditional/Simplified Chinese. This setting is in percentage.",
```
Comment (rxri) [inline CustomApps/lyrics-plus/Settings.js:490]: Don't use "Same as above [...]" wording

**Review by rxri — CHANGES_REQUESTED** [review]

File: CustomApps/lyrics-plus/Settings.js (hunk 4 above)
Comment (Deskehs) [reply CustomApps/lyrics-plus/Settings.js]: Done.

**Review by rxri — APPROVED** [review]

**Review by kyrie25 — APPROVED** [review]

**Comment by Aurora-Simba** [comment]
I am currently facing some problems in the lastest veision. When the subtitles are in traditional Chinese, the display mode is set to simplified Chinese, but the lyrics are still in traditional Chinese. Convert is already on, and the setting for traditional Chinese detection is 5%

**Pull request merged**